### ISSUE TITLE
`nfs_udp`, `nfs_version` not respected(?)

### ISSUE THREAD

**Issue by phs** [Learnosity/vagrant-nfs_guest#47]
If I add an `nfs_guest` synced folder as so:

```
  config.vm.synced_folder '/the/host/mount/path', '/the/guest/export/path', type: 'nfs_guest', nfs_udp: true, nfs_version: 4
```

..then it seems I still get a TCP version 3 mount:

```
$ nfsstat -v -c -m /the/host/mount/path
/the/host/mount/path from 192.168.33.10:/the/guest/export/path
  -- Original mount options:
     General mount flags: 0x10000000 noatime
     NFS parameters:
     File system locations:
       /the/guest/export/path @ 192.168.33.10 (192.168.33.10)
  -- Current mount parameters:
     General mount flags: 0x14000018 nodev,nosuid,noatime multilabel
     NFS parameters: vers=3,tcp,port=2049,nomntudp,hard,nointr,noresvport,negnamecache,callumnt,locks,quota,rsize=32768,wsize=32768,readahead=16,dsize=4096,nordirplus,nodumbtimr,timeo=10,maxgroups=16,acregmin=5,acregmax=60,acdirmin=5,acdirmax=60,nomutejukebox,nonfc,sec=sys
     File system locations:
       /the/guest/export/path @ 192.168.33.10 (192.168.33.10)
     fh 8 010001001f7d7e29
     Current location: 0x1 0 0 0:
       /the/guest/export/path @ 192.168.33.10 (192.168.33.10)
     Status flags: 0x0
```

Notice that the original mount options (`NFS parameters`) do not mention `udp` or `vers=4`.  If we also add those to `mount_options`:

```
  config.vm.synced_folder '/the/host/mount/path', '/the/guest/export/path', type: 'nfs_guest', nfs_udp: true, nfs_version: 4, mount_options: %w{udp vers=4}
```

..then we fail to mount, perhaps due to firewall issues:

```
$ vagrant reload
# ...
==> default: Exporting NFS shared folders from guest...
==> default: Setup firewall on the guest to allow NFS exports...
==> default: Preparing to edit /etc/exports on the guest...
# ...
==> default: Mounting NFS shared folders from guest...
    default: /mnt/fin-vm/disk => /Users/phs/.fin-vm/disk
mount_nfs: no usable addresses for host: 192.168.33.10
mount_nfs: no usable addresses for host: 192.168.33.10
mount_nfs: can't mount /the/guest/export/path from 192.168.33.10 onto /the/host/mount/path: No such file or directory
```

However if we omit `vers=4` from the explicit mount options:

```
  config.vm.synced_folder '/the/host/mount/path', '/the/guest/export/path', type: 'nfs_guest', nfs_udp: true, nfs_version: 4, mount_options: %w{udp}
```

...then we can actually get a v3 udp:

```
$ nfsstat -v -c -m /the/host/mount/path
/the/host/mount/path from 192.168.33.10:/the/guest/export/path
  -- Original mount options:
     General mount flags: 0x0
     NFS parameters: udp
     File system locations:
       /the/guest/export/path @ 192.168.33.10 (192.168.33.10)
  -- Current mount parameters:
     General mount flags: 0x4000018 nodev,nosuid multilabel
     NFS parameters: vers=3,udp,port=2049,nomntudp,hard,nointr,noresvport,conn,negnamecache,callumnt,locks,quota,rsize=8192,wsize=8192,readahead=16,dsize=4096,nordirplus,nodumbtimr,timeo=10,maxgroups=16,acregmin=5,acregmax=60,acdirmin=5,acdirmax=60,nomutejukebox,nonfc,sec=sys
     File system locations:
       /the/guest/export/path @ 192.168.33.10 (192.168.33.10)
     fh 8 010001001f7d7e29
     Current location: 0x1 0 0 0:
       /the/guest/export/path @ 192.168.33.10 (192.168.33.10)
     Status flags: 0x0
```

We get the same result if `nfs_udp` is `false`, making me think the option is wholly ignored:

```
  config.vm.synced_folder '/the/host/mount/path', '/the/guest/export/path', type: 'nfs_guest', nfs_udp: false, nfs_version: 4, mount_options: %w{udp}
```

**Comment by alangarf** [comment]
Hey thanks for the post.

TBH I'm not sure the plugin has much to do with this. It honestly passes the options through to the mount commands for the NFS mount. I've had a brief look at the code and I can't see what would be forcing you to be using v3 other than the daemon and the mount defaults itself.

Can you give me an idea of the OS you're using on the host and guest?

**Comment by phs** [comment]
Sure.  Host is Mac 10.12.6, guest is `centos/7` (box version `1801.2`).  The nfs server within the guest is manually brought in by `yum` and uses its defaults.

>  It honestly passes the options through to the mount commands for the NFS mount.

Except it seems it doesn't(?)  Even with the explicit `mount_options` option omitted, the `nfs_udp` `nfs_version` options appear to have no effect on the options passed to the host mount.  It's not a big problem since `mount_options` _does_ work.

**Comment by alangarf** [comment]
Ah ok, `nfs_udp` and `nfs_version` are for the "vagrant" NFS share plugin that's built-in. At the time this plugin was written these weren't easily accessible and hence `mount_options` was added. I'll see if I can modify the plugin to support them too. Thanks.

**Comment by phs** [comment]
Thanks.  If you didn't feel like supporting them, I'd be happy just with a bit of usage documentation.  The only thing I could find about how to use the plugin was [this comment][0].

After reading that, I went to vagrant's docs, and tried the options there, which is how I came across the problem.  A two paragraph blurb in the `README` would have saved the confusion.

[0]: https://github.com/Learnosity/vagrant-nfs_guest/issues/30#issuecomment-227065241
